base/channels.jl

```diff
@@ -275,12 +368,41 @@ function put_unbuffered(c::Channel, v)
         end
     end
     taker = popfirst!(c.takers)
-    yield(taker, v) # immediately give taker a chance to run, but don't block the current task
+    schedule(taker, v) # immediately give taker a chance to run but don't block the current task
     return v
 end
 
+end # !JULIA_PARTR
+
 push!(c::Channel, v) = put!(c, v)
 
+if JULIA_PARTR
+
+"""
+    fetch(c::Channel)
+
+Wait for and get the first available item from the channel. Does not
+remove the item. `fetch` is unsupported on an unbuffered (0-size) channel.
+"""
+function fetch(c::Channel)
+    c.sz_max == 0 && throw(ErrorException("`fetch` is not supported on an unbuffered Channel"))
```

Comment:
It would be good to change this to something other than `ErrorException` 
Since that type gives no useful information for catching purposes.
It is thrown a lot by user level (i.e. not pkg level) because it is the type of `error("Oh Noooo")`.